### PR TITLE
adding basic mm api k8s templates

### PR DESCRIPTION
**Pull request by adamkroonbcgov**
# Summary
Kubernetes Templates for Matchmaking API

[](https://hamzamohdzubair.github.io/redant/)

N/A

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] Dependency updates (may include refactoring of a dependencies functionality).
- [x] New feature (non-breaking change which adds functionality).

## How Has This Been Tested?
N/A

## Checklist:

- [x] I have read the checklist.

### CODE REVIEW

File: openshift/templates/matchmaking/api/mm-api-dc-temaple.yaml [hunk 1]
```diff
@@ -1 +1,97 @@
...
+          deploymentconfig: ${APPLICATION_NAME}
+        annotations:
+          openshift.io/generated-by: OpenShiftWebConsole
+      spec:
+        containers:
+          - name: ${APPLICATION_NAME}
+            image: >-
+              image-registry.openshift-image-registry.svc:5000/ec1236-tools/match-making-api@sha256:c3d57523a0277faacef24e3703b31f42dbfb3fbde88b3c27036886f7abaad78c
+            ports:
+              - containerPort: 8080
```
Comment (BradyMitch) [inline openshift/templates/matchmaking/api/mm-api-dc-temaple.yaml:65]: The matchmaking api container has a port of 8080?

**Review by BradyMitch — APPROVED** [review]
How are we using the route and service here? I know in PIMS we use a webhook linked to our route, service, and event listener to run the dev pipeline when changes are pushed in the repo

File: openshift/templates/matchmaking/api/mm-api-dc-temaple.yaml (hunk 1 above)
Comment (adamkroonbcgov) [reply openshift/templates/matchmaking/api/mm-api-dc-temaple.yaml:65]: default value of the container port

File: openshift/templates/matchmaking/api/mm-api-service-template.yaml [hunk 2]
```diff
@@ -1 +1,57 @@
...
+      openshift.io/generated-by: OpenShiftWebConsole
+  spec:
+    clusterIP: 10.98.235.174
+    ipFamilies:
+      - IPv4
+    ports:
+      - name: 8080-tcp
+        protocol: TCP
+        port: 8080
+        targetPort: 8080
```
Comment (Bottle7) [inline openshift/templates/matchmaking/api/mm-api-service-template.yaml:26]: Same as mentioned by @BradyMitch above.

**Review by Bottle7 — APPROVED** [review]

**Pull request merged**